core/src/main/scala/tech/beshu/ror/accesscontrol/blocks/rules/auth/BaseGroupsRule.scala

```diff
@@ -231,7 +232,8 @@ abstract class BaseGroupsRule(val settings: Settings)
     val externalAvailableGroups = sourceBlockContext.userMetadata.availableGroups
     for {
       externalGroupsMappedToLocalGroups <- mapExternalGroupsToLocalGroups(groupMappings, externalAvailableGroups)
-      availableLocalGroups <- calculateAllowedGroupsForUser(potentiallyAvailableGroups, PermittedGroupIds(externalGroupsMappedToLocalGroups))
+      potentiallyPermitted = GroupIds(externalGroupsMappedToLocalGroups)
+      availableLocalGroups <- GroupsLogic.Or(potentiallyPermitted).availableGroupsFrom(potentiallyAvailableGroups)
```

Comment:
This change restores an original implementation. 
- original implementation was an instersect on sets
- later it was modfied to use `calculateAllowedGroupsForUser` ot the specific rule (in my opinion it was not necessary; it was part of a change that introduced AND rule, but IMO it did not have any effect on the final result, tests detected no changes)
- restored in this PR as GroupsLogic.Or (because simple intersect, as in original impl will no longer work, much changed)
- this line needs second opinion